core/vtk/ttkSignedDistanceField/ttkSignedDistanceField.cpp

```diff
@@ -0,0 +1,262 @@
+    Origin[2] -= spacing_[2];
+  }
+}
+
+int ttkSignedDistanceField::RequestData(vtkInformation *ttkNotUsed(request),
+                                        vtkInformationVector **inputVector,
+                                        vtkInformationVector *outputVector) {
+  using ttk::SimplexId;
+
+  const auto domain = vtkDataSet::GetData(inputVector[0]);
+
+  if(!domain)
+    return !this->printErr("Unable to retrieve required input data object.");
+
+  // triangulation Domain
+  auto triangulation = ttkAlgorithm::GetTriangulation(domain);
+  if(!triangulation) {
+    this->printErr("Input triangulation pointer is NULL.");
+    return -1;
+  }
+
+  this->preconditionTriangulation(triangulation);
+
+  if(triangulation->isEmpty()) {
+    this->printErr("Triangulation allocation problem.");
+    return -2;
+  }
+
+  if(!domain) {
+    this->printErr("Input pointer is NULL.");
+    return -3;
+  }
+
+  const auto numberOfVertices = domain->GetNumberOfPoints();
+  if(numberOfVertices <= 0) {
+    this->printErr("Domain has no points.");
+    return -4;
+  }
+
+  std::stringstream ss;
+  ss << "Surface number of vertices : " << triangulation->getNumberOfVertices();
+  printMsg(ss.str());
+
+  ss.str("");
+  ss << "Surface number of triangles : "
+     << triangulation->getNumberOfTriangles();
+  printMsg(ss.str());
+
+  // Create encompassing triangulation
+  computeOutputInformation(inputVector);
+  vtkSmartPointer<vtkImageData> imageData
+    = vtkSmartPointer<vtkImageData>::New();
+  imageData->SetOrigin(Origin[0], Origin[1], Origin[2]);
+  imageData->SetSpacing(spacing_[0], spacing_[1], spacing_[2]);
+  imageData->SetDimensions(DataExtent[1], DataExtent[3], DataExtent[5]);
+
+  auto encompassingTriangulation = ttkAlgorithm::GetTriangulation(imageData);
+  if(!encompassingTriangulation) {
+    this->printErr("Input encompassing triangulation pointer is NULL.");
+    return -5;
+  }
+
+  if(encompassingTriangulation->isEmpty()) {
+    this->printErr("Encompassing triangulation allocation problem.");
+    return -6;
+  }
+
+  ss.str("");
+  ss << "Field number of vertices : "
+     << encompassingTriangulation->getNumberOfVertices();
+  printMsg(ss.str());
+
+  ss.str("");
+  ss << "Field number of edges : "
+     << encompassingTriangulation->getNumberOfEdges();
+  printMsg(ss.str());
+
+  // create output arrays
+  auto outputNoPoints = imageData->GetNumberOfPoints();
+
+  vtkNew<vtkFloatArray> outputScalars{};
+  outputScalars->SetNumberOfTuples(outputNoPoints);
+  outputScalars->SetName("signedDistanceField");
+
+  vtkNew<vtkIntArray> edgeCrossing{};
+  edgeCrossing->SetNumberOfTuples(outputNoPoints);
+  edgeCrossing->SetName("edgeCrossing");
+
+  vtkNew<vtkIntArray> isInterior{};
+  isInterior->SetNumberOfTuples(outputNoPoints);
+  isInterior->SetName("isInterior");
+
+  int ret{};
+
+  backend_ = Backend;
+  fastMarchingOrder_ = FastMarchingOrder;
+  ttkVtkTemplate2Macro(
+    outputScalars->GetDataType(), triangulation->getType(),
```

Comment:
just a side remark here (you don't need to change it), but in principle, you don't need to template on the output type (it's a constant, here, floats)

Reply:
I think I will modify it to reduce the compilation time.